### DISCUSSION TITLE
Best way to self-bottle for multiple architectures?

### DISCUSSION THREAD

**Discussion by sjml** [Homebrew/discussions#4137] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20-99-ge4c18be
ORIGIN: https://github.com/Homebrew/brew
HEAD: e4c18be302faf6209571c14ba2524bc4f9de7fe3
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9d2fca0f48c1a7966a141aad47456dfb92bc934c
Core tap last commit: 34 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_AUTO_UPDATE_SECS: 86400
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_EDITOR: vim
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_ANALYTICS: set
HOMEBREW_NO_ENV_HINTS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.1 => /opt/homebrew/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
﻿﻿Your system is ready to brew.
```

### Description of issue

(I know homebrew-core doesn't do cross-compiling or universal binaries; this question is strictly about how I could handle them in a third-party tap.)

I've got a personal tap where I'd like to offer bottled versions of the formulae -- it seems the best way to do that is with the standard `publish.yml` and `tests.yml`. But since GitHub doesn't yet have freely available M1 runners, and the workflows only make bottles for the systems on which they're running, it's not easy to make ARM bottles for newer Macs. 

With at least one of the tools I'm using Rust and could easily do a cross-compile or even make a universal executable, regardless of the architecture of the runner, but what's the recommended way to get that potential picked up by the automated bottling process? I could run a custom build process instead of just running `cargo install`, but how to label the bottle such that the automated edits to the formula would mark the bottles correctly? Is this even doable?

**Comment by SMillerDev** [comment, marked as answer]
Brew operations are heavily tied to the platform they run on so essentially everything is optimised against cross compiling. I think the easiest approach would be to compile the binaries and distribute them as a cask.

**Reply by sjml** [reply]
Ah, I suppose that makes sense. It's not the worst thing to always build from source, so maybe I'll just hold off until GitHub has Apple Silicon runners available. Thanks for the info!
